test/integration/fixtures/blocks/core__pullquote__custom-colors.serialized.html

```diff
@@ -1,3 +1,3 @@
 <!-- wp:pullquote {"textAlign":"right","className":"is-style-default","style":{"color":{"background":"#3486aa","text":"#ffefef"},"border":{"color":"#045679"}}} -->
-<figure class="wp-block-pullquote has-text-align-right is-style-default has-border-color has-text-color has-background" style="background-color:#3486aa;border-color:#045679;color:#ffefef"><blockquote><p>pullquote</p><cite>citation</cite></blockquote></figure>
+<figure class="wp-block-pullquote has-text-align-right is-style-default has-border-color has-text-color has-background" style="border-color:#045679;background-color:#3486aa;color:#ffefef"><blockquote><p>pullquote</p><cite>citation</cite></blockquote></figure>
```

Comment:
Just adding a brain dump here (there aren't any issues, but I was curious): I was wondering about the potential implications of the order of style values being rendered differently, given that this test fixture needed to be updated.

From manual testing of using the following markup, it looks like the block validation doesn't care about the order of the style values, just that the actual style is equal. Here's the markup I used with this PR applied (the version of this fixture prior to this PR):

```
<!-- wp:pullquote {"textAlign":"right","style":{"color":{"background":"#3486aa","text":"#ffefef"},"border":{"color":"#045679"}},"className":"is-style-default"} -->
<figure class="wp-block-pullquote has-text-align-right is-style-default has-border-color has-text-color has-background" style="background-color:#3486aa;border-color:#045679;color:#ffefef"><blockquote><p>pullquote</p><cite>citation</cite></blockquote></figure>
<!-- /wp:pullquote -->
```

With that markup copy / pasted into the code editor view with this PR applied, I logged out values from [`getStyleProperties`](https://github.com/wordpress/gutenberg/blob/03fd2b68e670cf3a071550db550458ba6c732000/packages/blocks/src/api/validation/index.js#L384) (where block validation splits the style string into an object of key / value pairs) to confirm that the keys of the object are in a different order for the actual vs expected versions of the block:

<img width="521" alt="image" src="https://user-images.githubusercontent.com/14988353/166850356-c87a2dca-ff40-468d-95dd-5234934ed9ee.png">

And then, the [isEqual check](https://github.com/wordpress/gutenberg/blob/03fd2b68e670cf3a071550db550458ba6c732000/packages/blocks/src/api/validation/index.js#L416) is happy about this since it's comparing objects instead of a simple string comparison. Phew! 😁

I mostly just wanted to confirm that we can happily change the order of rendering of style properties and not worry about having to add deprecations or anything like that for existing blocks / block supports (or worry about the order in which the style engine is invoked when generating styles). Good stuff!

Reply:
Thanks for double-checking @andrewserong 

I also was wondering about this (and pasted the previous order in the code editor too), but fortunately you had already done most of the hard work in https://github.com/WordPress/gutenberg/pull/39039 !

That change saved me a lot of time, so thank you then and now 🙇